packages/dataviews/src/dataviews.js

```diff
@@ -42,11 +48,32 @@ export default function DataViews( {
 		( v ) => v.type === view.type
 	).component;
 	const _fields = useMemo( () => {
-		return fields.map( ( field ) => ( {
-			...field,
-			render: field.render || field.getValue,
-		} ) );
+		return fields.map( ( field ) => {
+			// Normalize formats.
+			field.formats = field.formats || [];
+
+			// Normalize render.
+			switch ( field.type ) {
+				case DATE_TYPE:
+					field.render = ( { item } ) =>
```

Comment:
It's not clear to me why we're updating the "field" here instead of doing something like:


```
const fieldType = getFieldType( field.type };
fieldType.render();
```

It seems to me that it's better to keep things clear as otherwise, we'll be wondering is this "field" the full field or the one with optional stuff...